### ISSUE TITLE
Allow iframing the launch tab on localhost

### ISSUE THREAD

**Issue by yaustar** [playcanvas/editor#668]
This would help developers workflows who are using PlayCanvas as part of a larger webapp such as within a react website or similar.

**Comment by tonetechnician** [comment]
Yeah, +1 for this! Exactly what is required for me to enhance developer flow

**Comment by slimbuck** [comment]
@yak32 presumably this requires just a slightly different content policy rule on our end?

**Comment by Icosaplex** [comment]
This would be really helpful

**Comment by DeaconDP** [comment]
+1

**Comment by tobowers** [comment]
+1 - our app has a somewhat complex relationship with the hosting page (requesting things like sending transactions, etc). And our host page puppets the game as well. I can do most of the testing using a test harness within the app, but the last mile of testing gets annoying with having to do builds all the time.

**Comment by willeastcott** [comment]
Hey @yak32 - can you give an update (when you're back) on the feasibility of supporting this?

**Comment by yak32** [comment]
> Hey @yak32 - can you give an update (when you're back) on the feasibility of supporting this?

I need to discuss it with our security team.

**Comment by yak32** [comment]
our launch page uses authorisation cookie to authenticate requests (only authorised users are allowed to preview the current project). Cookie's SameSite attribute is set to 'Lax' to only allow same domain to receive an authorisation cookies for security reasons. Web pages in iframe only sends cookies with SameSite set to 'none'. Unfortunately we can't change our auth cookies to have SameSite 'none' for security reasons, so I don't see an easy solution without significant redesign of our authentication method for now.

**Comment by yak32** [comment]
one potential solution could be uploading some html as an asset, and embed a launch page there.

**Comment by yaustar** [comment]
I wonder if there is a Chrome extension that could be installed to force cross domain cookies for local development?

**Comment by yaustar** [comment]
Facebook (for instant games) have the ability to create test accounts that have read access permissions to the project. 

It be a bigger job than I would like: related to https://github.com/playcanvas/editor/issues/791, a team/user could create a read only test user account that has a lower security cookie for read only access?

**Comment by yaustar** [comment]
What do you mean @tobowers ?

**Comment by tobowers** [comment]
> What do you mean by local server @tobowers ?

I mean "npm run local-playcanvas-dev --project-id=abc"

**Comment by antoinemacia** [comment]
@yaustar  this is pretty sound and also a common implementation for micro frontends;

One relatively simple pattern I see in SDKs that allow third party extension (where the endpoint is unauthenticated like the launch url), is to allow users to specify domains that would host PC launch links via the admin, and check the application key matches in the iframe URL

Would that comply to PC's security policies?
